src/plugins/service-worker/index.js

```diff
@@ -0,0 +1,45 @@
+// @flow
+const buildPaths = require('mwp-cli/src/commands/buildCommands/config/paths');
+const path = require('path');
+
+/*
+ * The service worker plugin will register a root-level route for serving the
+ * service worker script file.
+ *
+ * Disk location of the script file is determined by builder CLI config import.
+ */
+export default function register(
+	server: Object,
+	options: ?Object,
+	next: () => void
+) {
+	/*
+	 * Route for service worker script at top-level path. Depends on `Inert`
+	 * `path` must match client `serviceWorker.register` call - MWP provides this
+	 * in the `<ServiceWorker>` component
+	 */
+	server.route({
+		method: 'GET',
+		path: '/asset-service-worker.{localeCode}.js',
```

Comment:
The original ticket was implementing a webpack offline plugin, so the implementation is still well-aligned with that that goal - I've included comments to update what I learned when investigating options https://meetup.atlassian.net/browse/WP-321

> I suspect we don't want to front load all assets for our site, at least not in the long term

With our current assets, it's not an issue because it's only a few hundred kB of assets (much less than the image payload for many of our pages), but we can definitely tweak [the  plugin config](https://github.com/meetup/mwp-cli/blob/270cb2bbd84d8455c695e1583c1798f1ac73e0e8/src/commands/buildCommands/config/webpack/browserAppConfig.js#L143-L152) as desired over time to tune performance.

It might be worth looking up, but I'm pretty sure the service worker cache is effectively the same as the browser cache, so we don't need to manually manage purging it ourselves.

Since the service worker will only run in prod and there is no consumer-app code needed to make it work, I think we don't need to worry too much about confusion for developers. However, I will document some of this behavior in the caching docs that I put together this sprint.